src/otx/algorithms/anomaly/adapters/anomalib/callbacks/xpu.py

```diff
@@ -0,0 +1,39 @@
+"""Anomaly XPU device callback."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import torch
+from pytorch_lightning import Callback
+
+from otx.algorithms.common.utils.utils import is_xpu_available
+
+
+class XPUCallback(Callback):
+    """XPU device callback.
+
+    Applies IPEX optimization before training, moves data to XPU.
+    """
+
+    def __init__(self, device_idx=0):
+        self.device = torch.device(f"xpu:{device_idx}")
+
+    def on_fit_start(self, trainer, pl_module):
+        """Applies IPEX optimization before training."""
+        if is_xpu_available():
+            pl_module.to(self.device)
+            model, optimizer = torch.xpu.optimize(trainer.model, optimizer=trainer.optimizers[0], dtype=torch.float32)
```

Comment:
(minor) If we use fp32 model, we don't need to assign `dtype` arg.

Reply:
It's a big question if we should always rely on default arg values or not in general. Implicit dtype could be useful for understanding that this callback works only with fp32 for those who're not that closely familiar to IPEX API.